### PR TITLE
Show users the date as they left the agenda when users return to the page

### PR DESCRIPTION
**Pull request by MaksymShokin**
@tonytlwu @squallstar 

## Issue
Fliplet/fliplet-studio#4973

## Description
Added page context update. If the user goes to the default page context is removed.

## Screenshots/screencasts
agenda demo

## Backward compatibility
This change is fully backward compatible.

### CODE REVIEW

File: js/layout-javascript/agenda-code.js [hunk 1]
```diff
@@ -294,10 +294,13 @@ DynamicList.prototype.attachObservers = function() {
           return;
         }
 
+        var defaultDateIndex = _this.getDateIndex(_this.Utils.Date.moment());
         var indexOfActiveDate = _this.$container.find('.agenda-date-selector li').not('.placeholder').index(_this.$container.find('.agenda-date-selector li.active'));
         var indexOfClickedDate = _this.$container.find('.agenda-date-selector li').not('.placeholder').index(_this.$container.find('.agenda-date-selector li.active').next());
         var indexDifference = indexOfClickedDate - indexOfActiveDate;
 
+        _this.updateNavigatorQuery(defaultDateIndex, indexOfClickedDate)
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]: Missing semicolon

File: js/layout-javascript/agenda-code.js [hunk 2]
```diff
@@ -306,10 +309,13 @@ DynamicList.prototype.attachObservers = function() {
           return;
         }
 
+        var defaultDateIndex = _this.getDateIndex(_this.Utils.Date.moment());
         var indexOfActiveDate = _this.$container.find('.agenda-date-selector li').not('.placeholder').index(_this.$container.find('.agenda-date-selector li.active'));
         var indexOfClickedDate = _this.$container.find('.agenda-date-selector li').not('.placeholder').index(_this.$container.find('.agenda-date-selector li.active').prev());
         var indexDifference = indexOfClickedDate - indexOfActiveDate;
 
+        _this.updateNavigatorQuery(defaultDateIndex, indexOfClickedDate)
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]: Missing semicolon

File: js/layout-javascript/agenda-code.js [hunk 3]
```diff
@@ -321,10 +327,13 @@ DynamicList.prototype.attachObservers = function() {
         return;
       }
 
+      var defaultDateIndex = _this.getDateIndex(_this.Utils.Date.moment());
       var indexOfActiveDate = _this.$container.find('.agenda-date-selector li').not('.placeholder').index(_this.$container.find('.agenda-date-selector li.active'));
       var indexOfClickedDate = _this.$container.find('.agenda-date-selector li').not('.placeholder').index(this);
       var indexDifference = indexOfClickedDate - indexOfActiveDate
 
+      _this.updateNavigatorQuery(defaultDateIndex, indexOfClickedDate)
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]: Missing semicolon

File: js/layout-javascript/agenda-code.js [hunk 4]
```diff
@@ -509,6 +518,16 @@ DynamicList.prototype.scrollEvent = function() {
   });
 }
 
+DynamicList.prototype.updateNavigatorQuery = function(defaultDateIndex, indexOfClickedDate) {
+  if (defaultDateIndex === indexOfClickedDate) {
+    Fliplet.Page.Context.remove('dayUserLeft');
+  } else {
+    Fliplet.Page.Context.update({
+      dayUserLeft: indexOfClickedDate
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]: Please use the `dateIndex` key. This means we also add the capability of taking users to a date via index.

```suggestion
      dateIndex: indexOfClickedDate
```

File: js/layout-javascript/agenda-code.js [hunk 5]
```diff
@@ -509,6 +518,16 @@ DynamicList.prototype.scrollEvent = function() {
...
+    });
+  }
+}
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]: Missing semicolon

File: js/layout-javascript/agenda-code.js [hunk 6]
```diff
@@ -1567,14 +1586,26 @@ DynamicList.prototype.getDateIndex = function (date) {
...
+  this.goToDate(moment().format('YYYY-MM-DD'), false);
 };
 
-DynamicList.prototype.goToDate = function (date) {
+DynamicList.prototype.goToDate = function (date, goToDayUserLeft) {
   if (!date) {
     return;
   }
 
+  if (goToDayUserLeft) {
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]: The variable name doesn't explain very well what its purpose is. Please explain.

Comment (MaksymShokin) [reply js/layout-javascript/agenda-code.js]: @tonytlwu I added this variable so function uses information from the navigator.query which is the id of saved day or the  default date which comes in the YYYY-MM-DD format.

File: js/layout-javascript/agenda-code.js [hunk 7]
```diff
@@ -1567,14 +1586,26 @@ DynamicList.prototype.getDateIndex = function (date) {
 };
 
 DynamicList.prototype.goToToday = function () {
-  this.goToDate(moment().format('YYYY-MM-DD'));
+  var dayUserLeft = parseInt(Fliplet.Navigate.query.dayUserLeft, 10);
+
+  if (dayUserLeft) {
+    this.goToDate(dayUserLeft, true);
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]: Why not just use:

```suggestion
    this.sliderGoTo(dayUserLeft);
```

Then there's no need to modify how `this.goToDate` works.

File: js/layout-javascript/agenda-code.js (hunk 2 above)
Comment (MaksymShokin) [reply js/layout-javascript/agenda-code.js]: Fixed

File: js/layout-javascript/agenda-code.js [hunk 8]
```diff
@@ -294,10 +294,13 @@ DynamicList.prototype.attachObservers = function() {
           return;
         }
 
+        var defaultDateIndex = _this.getDateIndex(_this.Utils.Date.moment());
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]: This seems to be used exclusively in the `updateNavigatorQuery()` function. Can we define this variable inside the `updateNavigatorQuery()` function instead?

File: js/layout-javascript/agenda-code.js [hunk 9]
```diff
@@ -509,6 +518,16 @@ DynamicList.prototype.scrollEvent = function() {
   });
 }
 
+DynamicList.prototype.updateNavigatorQuery = function(defaultDateIndex, indexOfClickedDate) {
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]: - Function name change
- Removes unnecessary parameter

```suggestion
DynamicList.prototype.updateDateIndexContext = function(indexOfClickedDate) {
```

File: js/layout-javascript/agenda-code.js (hunk 8 above)
Comment (MaksymShokin) [reply js/layout-javascript/agenda-code.js]: Fixed

File: js/layout-javascript/agenda-code.js (hunk 9 above)
Comment (MaksymShokin) [reply js/layout-javascript/agenda-code.js]: Fixed

**Pull request merged**